// lottie-converter/index.js
const axios = require("axios")
const FormData = require("form-data");

async function main(options) {
    if (!options.file) throw "File cannot be empty."
    if (!options.format) throw "Format cannot be empty."
    const form = new FormData()
    form.append(`file`, Buffer.from(options.file), {
        filename: `${options.filename ?? 'new.json'}`,
    });
    if(options.format !== 'lottie') {
        form.append(`format`, options.format);
        form.append(`base64_response`, 1);
        form.append(`width`, options.width ?? 1000);
        form.append(`height`, options.height ?? 1000);
    }
    let r = await axios({
        method: 'post',
        url: options.format === 'lottie' ? 'https://iconscout.com/lottie-optimizer?base64_response=1&format=dotlottie' : `https://iconscout.com/lottie-download?`,
        data: form,
        headers: {
            'Content-Type': 'multipart/form-data',
        },
    }).catch(function (error) {
        if (error.response) {
            throw new Error(JSON.stringify({
                statusCode: error.response.status,
                data: error.response.data
            }, null, 4))
        } else {
            throw new Error("Opse something unknown happened! :(")
        }
    })
    return r.data
}
module.exports = main
